### PR TITLE
Remove null values from servers in the same way we were doing it for …

### PR DESCRIPTION
**Pull request by chrisbaldauf**
…deployments. Move the deletion logic to the model. Still need to rationalize enforcing required fields on post and put

@maclennann @potashj

### CODE REVIEW

File: api/controllers/deployments.js [hunk 1]
```diff
@@ -21,10 +21,12 @@ function getDeployments(req, res, next) {
     offset: req.swagger.params.offset.value,
     order: [ ["createdAt", "DESC"] ]
   }).then(function(deployments) {
-    for(var i = 0; i < deployments.length; i++) {
-      var deployment = deployments[i];
-      deleteNullValues(deployment.dataValues);
-    }
+    deployments.map(function(deployment) {
```
Comment (chrisbaldauf) [inline api/controllers/deployments.js]: Change to map syntax and added deleteNullValues for each server.

File: test/api/controllers/servers.js [hunk 2]
```diff
@@ -40,9 +40,7 @@ describe("controllers", function() {
       var testServer = {
         deployment_id: deployment_id,
         hostname: hostname,
-        ip_address: "127.0.0.1",
-        result: "success",
```
Comment (chrisbaldauf) [inline test/api/controllers/servers.js:44]: Remove result and elapsed_seconds for the POST, so that we can test it as we're expecting to use it.

**Comment by ryanbreen** [comment]
Should the deletion logic be a loop to kill any prop with a null value? That seems less annoying than needing to add delete logic for any new prop.

**Comment by chrisbaldauf** [comment]
Yeah, I was working my way there. Just wanted to make sure that this would
plumb through properly. Ideally, I'd like to make all of the other fields
(besides result and elapsed_seconds) required, but I'm not sure how to do
that while still allowing a PUT with only those two optional fields.

File: api/controllers/deployments.js (hunk 1 above)
Comment (maclennann) [inline api/controllers/deployments.js]: Does `map` act any differently than `forEach` given that you aren't assigning the return values to a new array?

**Pull request merged**